src/ethereum_wallet.rs

```diff
@@ -0,0 +1,209 @@
+use crate::{geth, Seed};
+use comit::{
+    actions::ethereum::{CallContract, DeployContract},
+    asset::Erc20,
+    ethereum::{Address, ChainId, Hash, TransactionReceipt},
+};
+use reqwest::Url;
+
+#[derive(Debug, Clone)]
+pub struct Wallet {
+    private_key: clarity::PrivateKey,
+    geth_client: geth::Client,
+}
+
+impl Wallet {
+    pub fn new(seed: Seed, url: Url) -> anyhow::Result<Self> {
+        let private_key = clarity::PrivateKey::from_slice(&seed.into_inner())
+            .map_err(|_| anyhow::anyhow!("Failed to derive private key from slice"))?;
+
+        let geth_client = geth::Client::new(url);
+
+        Ok(Self {
+            private_key,
+            geth_client,
+        })
+    }
+
+    pub fn new_from_private_key(private_key: clarity::PrivateKey, url: Url) -> Self {
+        let geth_client = geth::Client::new(url);
+
+        Self {
+            private_key,
+            geth_client,
+        }
+    }
+
+    pub fn account(&self) -> anyhow::Result<Address> {
+        let address = self
+            .private_key
+            .to_public_key()
+            .map_err(|_| anyhow::anyhow!("Failed to derive public key from private key"))?;
+
+        Ok(Address::from_slice(&address.as_bytes()))
+    }
+
+    pub async fn deploy_contract(
+        &self,
+        DeployContract {
+            data,
+            gas_limit,
+            chain_id,
+            ..
+        }: DeployContract,
+    ) -> anyhow::Result<Hash> {
+        let nonce = self.get_transaction_count().await?;
+
+        let transaction = clarity::Transaction {
+            nonce: nonce.into(),
+            gas_price: 0u32.into(),
```

Comment:
What happen when gas price is 0?

Reply:
It works on a local development node, but it would never get mined on mainnet. Should I use different values depending on the network? Should it be a constant?

Reply:
Well, I think it's more a proper todo (task in basecamp) to decide on network fees using an external service.